native/core/src/models/status/membership.rs

```diff
@@ -1,147 +1,162 @@
-use matrix_sdk::ruma::{
-    events::room::member::MembershipChange as RumaMembershipChange, OwnedMxcUri, OwnedUserId,
+use matrix_sdk_base::ruma::{OwnedMxcUri, OwnedUserId};
+use matrix_sdk_ui::timeline::{
+    MemberProfileChange, MembershipChange as SdkMembershipChange, RoomMembershipChange,
 };
 use serde::{Deserialize, Serialize};
 
-/// A simple representation of a change, containing old and new data.
+// ruma_events::room::member::change::Change doesn't support serialization
 #[derive(Clone, Debug, Serialize, Deserialize)]
-#[allow(clippy::exhaustive_structs)]
 pub struct Change<T> {
     /// The old data.
-    pub old: T,
+    pub old_val: T,
 
     /// The new data.
-    pub new: T,
+    pub new_val: T,
 }
 
-/// Translation of the membership change in `m.room.member` event.
-#[derive(Clone, Debug, Serialize, Deserialize)]
-pub enum MembershipChangeType {
-    /// User joined the room.
-    Joined,
-
-    /// User left the room.
-    Left,
-
-    /// User was banned.
-    Banned,
-
-    /// User was unbanned.
-    Unbanned,
-
-    /// User was kicked.
-    Kicked,
-
-    /// User was invited.
-    Invited,
-
-    /// User was kicked and banned.
-    KickedAndBanned,
+impl<T: PartialEq> Change<T> {
+    pub fn new(old_val: T, new_val: T) -> Option<Self> {
+        if old_val == new_val {
+            None
+        } else {
+            Some(Self { old_val, new_val })
+        }
+    }
+}
 
-    /// User accepted the invite.
-    InvitationAccepted,
+#[derive(Clone, Debug, Serialize, Deserialize)]
+pub struct MembershipChange {
+    pub(crate) user_id: OwnedUserId,
+    pub(crate) change: Option<String>,
+}
 
-    /// User rejected the invite.
-    InvitationRejected,
+impl From<&RoomMembershipChange> for MembershipChange {
+    fn from(value: &RoomMembershipChange) -> Self {
+        MembershipChange {
+            user_id: value.user_id().to_owned(),
+            change: value.change().map(|c| {
```

Comment:
why remove the neatly organized and typed enum and replace it with  less easy to enforce `String` here? what's the benefit?